lambdas/build-cri-oauth-request/src/main/java/uk/gov/di/ipv/core/buildcrioauthrequest/BuildCriOauthRequestHandler.java

```diff
@@ -270,10 +270,32 @@ private JWEObject signEncryptJar(
             throws HttpResponseExceptionWithErrorBody, ParseException, JOSEException {
         SharedClaimsResponse sharedClaimsResponse =
                 getSharedAttributes(ipvSessionItem, userId, currentVcStatuses, criId);
-        EvidenceRequest evidence = null;
-        if (credentialIssuerConfigService.enabled(CoreFeatureFlag.EVIDENCE_REQUEST_ENABLED)) {
-            // Temp hard coded until we have logic to store it in session.
-            evidence = new EvidenceRequest(4);
+        EvidenceRequest evidenceRequest = null;
+        if (credentialIssuerConfigService.enabled(CoreFeatureFlag.EVIDENCE_REQUEST_ENABLED)
+                && criId.equals(F2F_CRI)) {
+            List<RequiredGpg45ScoresDto> requiredGpg45Scores =
+                    ipvSessionItem.getRequiredGpg45Scores();
+
+            int strengthScore = 0;
+            for (RequiredGpg45ScoresDto gpg45Score : requiredGpg45Scores) {
+                Gpg45ScoresDto requiredScores = gpg45Score.getRequiredScores();
+                if (requiredScores.getFraud() > 0) {
+                    continue;
+                }
+                List<EvidenceDto> evidences = requiredScores.getEvidences();
+                int minRequiredStrengthScore =
+                        evidences.stream()
+                                .mapToInt(EvidenceDto::getStrength)
+                                .summaryStatistics()
+                                .getMin();
```

Comment:
I think this is not quite right. When there's a list of evidence eg. [EV_32, EV_42] it means 2 pieces of evidence are required, one strength 3 the other 4. However for F2F and the profiles we're dealing with there should only ever be 1 piece of evidence in the list - so I think we can just assume and just get the strength score of the first piece of evidence. So something like `int minRequiredStrength = evidences.get(0).getStrength()`